### PR TITLE
Handle tile grids of TileWMS layers on reprojection

### PR DESCRIPTION
**Pull request by hblitza**
## Bug
ProjectionSwitcher in footer didn't work as all TileWMS layers weren't reprojected correctly.
## Fix
Use Ol function `setTileGridForProjection` on TileWMS layer source **before** layer is added to map.
## Future improvements
Get list of predefined crs for the projection switcher and apply the new function for each crs.

__

Thanks @dnlkoch  for co-working.

Please note: @terrestris/devs

### CODE REVIEW

File: src/util/AppContextUtil/Shogun2AppContextUtil.tsx [hunk 1]
```diff
@@ -291,14 +293,29 @@ class Shogun2AppContextUtil extends BaseAppContextUtil implements AppContextUtil
     const layerSource = new OlTileWMS({
       url: url,
       attributions: attribution,
-      tileGrid: tileGrid,
+      tileGrid: baseTileGrid,
+      projection: 'EPSG:3857',
```
Comment (hblitza) [inline src/util/AppContextUtil/Shogun2AppContextUtil.tsx:297]: The projection should be the mapView's projection by default. But is is always null. If specified here, the code works - but this cannot be the final solution. Ideas?

File: src/util/AppContextUtil/Shogun2AppContextUtil.tsx [hunk 2]
```diff
@@ -291,14 +293,29 @@ class Shogun2AppContextUtil extends BaseAppContextUtil implements AppContextUtil
...
+    // Following list could be set from AppContext in future
+    const crsList = ['EPSG:4326'];
+    // Create and define further tile grids
+    crsList.forEach(crs => {
+      if (crs === 'EPSG:3857') {
+        return;
+      }
+      const newTileGrid = this.createTileGridForProjection(baseTileGrid, crs);
+      if (newTileGrid && newTileGrid instanceof OlTileGrid) {
+        layerSource.setTileGridForProjection('EPSG:4326', newTileGrid);
```
Comment (dnlkoch) [inline src/util/AppContextUtil/Shogun2AppContextUtil.tsx:316]:
```suggestion
        layerSource.setTileGridForProjection(crs, newTileGrid);
```

File: src/util/AppContextUtil/Shogun2AppContextUtil.tsx (hunk 1 above)
Comment (dnlkoch) [reply src/util/AppContextUtil/Shogun2AppContextUtil.tsx:297]: Any chance to get the value from the given crs defined for the current application?

Comment (hblitza) [reply src/util/AppContextUtil/Shogun2AppContextUtil.tsx]: Maybe to store it as private property within the class? see new changes. Not sure though

Comment (dnlkoch) [reply src/util/AppContextUtil/Shogun2AppContextUtil.tsx]: This is fine for me.

But please check lint warnings.

Comment (hblitza) [reply src/util/AppContextUtil/Shogun2AppContextUtil.tsx]: yup, thanks for review

**Pull request merged**